server/webhook.go

```diff
@@ -123,11 +172,213 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, body
+			return
+		}
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(post); err != nil {
+		p.API.LogWarn("failed to write response", "error", err.Error())
+	}
+}
+
+func (p *Plugin) handleRecordingCompleted(w http.ResponseWriter, r *http.Request, body []byte) {
+	var webhook zoom.RecordingWebhook
+	if err := json.Unmarshal(body, &webhook); err != nil {
+		p.API.LogError("Error unmarshaling meeting webhook", "err", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	meetingPostID := webhook.Payload.Object.UUID
+	postID, appErr := p.fetchMeetingPostID(meetingPostID)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+
+	post, appErr := p.API.GetPost(postID)
+	if appErr != nil {
+		p.API.LogWarn("Could not get meeting post by id", "err", appErr)
+		http.Error(w, appErr.Error(), appErr.StatusCode)
+		return
+	}
+
+	recordings := make(map[time.Time][]zoom.RecordingFile)
+
+	for _, recording := range webhook.Payload.Object.RecordingFiles {
+		if recording.RecordingType == zoom.RecordingTypeChat {
+			recordings[recording.RecordingStart] = append(recordings[recording.RecordingStart], recording)
+		}
+
+		if recording.RecordingType == zoom.RecordingTypeVideo {
+			recordings[recording.RecordingStart] = append(recordings[recording.RecordingStart], recording)
+		}
+	}
+
+	for _, recordingGroup := range recordings {
+		newPost := &model.Post{
+			UserId:    p.botUserID,
+			ChannelId: post.ChannelId,
+			RootId:    post.Id,
+			Message:   "",
+			FileIds:   []string{},
+		}
+		for _, recording := range recordingGroup {
+			if recording.RecordingType == zoom.RecordingTypeChat {
+				request, err := http.NewRequest(http.MethodGet, recording.DownloadURL, nil)
+				if err != nil {
+					p.API.LogWarn("Unable to get the chat", "err", err)
+					http.Error(w, err.Error(), http.StatusBadRequest)
+					return
+				}
+				request.Header.Set("Authorization", "Bearer "+webhook.DownloadToken)
+				response, err := http.DefaultClient.Do(request)
+				if err != nil {
+					p.API.LogWarn("Unable to get the chat", "err", err)
+					http.Error(w, err.Error(), http.StatusBadRequest)
+					return
+				}
+				defer response.Body.Close()
+				chat, err := io.ReadAll(response.Body)
+				if err != nil {
+					p.API.LogWarn("Unable to get the chat", "err", err)
+					http.Error(w, err.Error(), http.StatusBadRequest)
+					return
+				}
+				fileInfo, appErr2 := p.API.UploadFile(chat, post.ChannelId, "Chat-history.txt")
+				if appErr2 != nil {
+					p.API.LogWarn("Unable to get the chat", "err", appErr2)
+					http.Error(w, appErr2.Error(), http.StatusBadRequest)
+					return
+				}
+
+				newPost.FileIds = append(newPost.FileIds, fileInfo.Id)
+				newPost.AddProp("captions", []any{map[string]any{"file_id": fileInfo.Id}})
+				newPost.Type = "custom_zoom_chat"
+				if newPost.Message == "" {
+					newPost.Message = " "
+				}
```

Comment:
Not sure what this is necessary for?

Reply:
This is no longer needed. It was there because I didn't fully understand that the chat history needed to be attached to a recording. I can remove it now.